src/test/worker/worker_server_proxy.py

```diff
@@ -0,0 +1,118 @@
+from ja.worker.proxy.proxy import IWorkerServerProxy
+from ja.server.database.types.work_machine import WorkMachine, WorkMachineState, WorkMachineResources
+from ja.common.job import Job, JobStatus, JobPriority, JobSchedulingConstraints
+from ja.common.docker_context import DockerContext, DockerConstraints
+from ja.common.message.server import ServerResponse
+from ja.common.work_machine import ResourceAllocation
+from typing import Dict
+import copy
+from unittest import TestCase
+from test.abstract import skipIfAbstract
+
+
+class WorkerServerProxyDummy(IWorkerServerProxy):
+    def __init__(self, wmcs: Dict[str, WorkMachine], jobs: Dict[str, Job]):
+        self._wmcs: Dict[str, WorkMachine] = copy.deepcopy(wmcs)
+        self._jobs: Dict[str, Job] = copy.deepcopy(jobs)
+
+    def register_self(self, uid: str, work_machine_resources: WorkMachineResources) -> ServerResponse:
+        if uid in self._wmcs:
+            return ServerResponse("work machine already exist!", False)
+        self._wmcs[uid] = WorkMachine(
+            uid=uid,
+            state=WorkMachineState.ONLINE,
+            resources=work_machine_resources
+        )
+        return ServerResponse("work machine added!", True)
+
+    def unregister_self(self, uid: str) -> ServerResponse:
+        if uid in self._wmcs:
+            self._wmcs[uid].state = WorkMachineState.RETIRED
+            return ServerResponse("work machine unregistred!", True)
+        return ServerResponse("work machine does not exist!", False)
+
+    def notify_job_finished(self, uid: str) -> ServerResponse:
+        if uid in self._jobs:
+            self._jobs[uid].status = JobStatus.DONE
```

Comment:
This will raise a ValueError if the method is called twice for the same Job.
A test for this would be nice to have, but not required.